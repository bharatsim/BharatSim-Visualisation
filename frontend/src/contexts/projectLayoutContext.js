import React from 'react';

const projectLayoutContext = React.createContext({
  projectMetadata: {
    id: undefined,
    name: undefined,
  },
  selectedDashboardMetadata: {
    name: undefined,
    _id: undefined,
  },
  addDashboard: null,
  deleteDashboard: null,
});

const ProjectLayoutProvider = projectLayoutContext.Provider;

export { projectLayoutContext, ProjectLayoutProvider };
