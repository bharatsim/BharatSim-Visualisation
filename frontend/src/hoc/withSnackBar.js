import React from 'react';
import { SnackbarProvider } from 'notistack';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';
import failureIcon from '../assets/images/error-icon.svg';
import successIcon from '../assets/images/success-icon.svg';

const useSnackBarStyles = makeStyles((theme) => ({
  contentRoot: {
    padding: theme.spacing(1, 4),
  },
  message: {
    padding: 0,
  },
  variantSuccess: {
    backgroundColor: `#EAF4E5 !important`,
    borderColor: `${theme.palette.success.main}33`,
    border: '1px solid',
    color: `${theme.palette.success.dark} !important`,
    ...theme.typography.body2,
    borderRadius: theme.spacing(1),
    boxShadow: 'unset',
    opacity: 1,
    minWidth: theme.spacing(120),
  },
  variantError: {
    backgroundColor: `#FBE9E4 !important`,
    borderColor: `${theme.palette.error.light}33`,
    border: '1px solid',
    color: `${theme.palette.error.dark} !important`,
    ...theme.typography.body2,
    borderRadius: theme.spacing(1),
    boxShadow: 'unset',
    opacity: 1,
    minWidth: theme.spacing(120),
  },
}));

const useIconStyles = makeStyles((theme) => ({
  iconRoot: {
    marginRight: theme.spacing(2),
    height: theme.spacing(10),
    width: theme.spacing(10),
  },
}));

function withSnackBar(WrappedComponent) {
  return (props) => {
    const snackBarClasses = useSnackBarStyles();
    const iconClasses = useIconStyles();
    const snackBarRef = React.createRef();
    const onClickDismiss = (key) => () => {
      snackBarRef.current.closeSnackbar(key);
    };
    return (
      <SnackbarProvider
        maxSnack={3}
        ref={snackBarRef}
        anchorOrigin={{ horizontal: 'right', vertical: 'bottom' }}
        classes={snackBarClasses}
        iconVariant={{
          error: <img src={failureIcon} alt="error" className={iconClasses.iconRoot} />,
          success: <img src={successIcon} alt="success" className={iconClasses.iconRoot} />,
        }}
        action={(key) => <Button onClick={onClickDismiss(key)}>Dismiss</Button>}
      >
        <WrappedComponent {...props} />
      </SnackbarProvider>
    );
  };
}

export default withSnackBar;
