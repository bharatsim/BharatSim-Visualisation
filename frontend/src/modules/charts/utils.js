import { chartStyleConfig, getColor } from './chartStyleConfig';

function trasformDataForChart(csvData, xColumn, yColumns) {
  return {
    labels: csvData && csvData.data[xColumn],
    datasets:
      csvData &&
      yColumns.map((yColumn, index) => {
        return {
          ...chartStyleConfig,
          label: yColumn,
          borderColor: getColor(index),
          backgroundColor: getColor(index),
          data: csvData && csvData.data[yColumn],
        };
      }),
  };
}

function getYaxisNames(yColumns) {
  return yColumns.map((yColumn) => yColumn.name);
}

export { trasformDataForChart, getYaxisNames };
