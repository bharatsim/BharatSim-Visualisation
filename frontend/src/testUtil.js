// eslint-disable-next-line import/no-extraneous-dependencies
import { fireEvent, within } from '@testing-library/react';

export function selectDropDownOption(container, dropDownId, optionId) {
  const dropDown = container.getByTestId(dropDownId);
  fireEvent.mouseDown(within(dropDown).getByRole('button'));
  const options = within(
    within(document.getElementById(`menu-${dropDownId}`)).getByRole('listbox'),
  );
  fireEvent.click(options.getByTestId(`${dropDownId}-${optionId}`));
}
