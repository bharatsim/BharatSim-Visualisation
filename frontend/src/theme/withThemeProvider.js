import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme(props) {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent {...props} />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
