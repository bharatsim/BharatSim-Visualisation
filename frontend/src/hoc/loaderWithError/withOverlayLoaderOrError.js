import React, { useState } from 'react';
import { OverlayLoaderOrErrorContextProvider } from '../../contexts/overlayLoaderOrErrorContext';
import OverlayLoader from '../../uiComponent/OverlayLoader';
import OverlayError from '../../uiComponent/OverlayError';
import {useDispatch, useSelector} from "react-redux";
import * as actions from "./actions";

function withOverlayLoaderOrError(WrappedComponent) {
  return (props) => {
    const loaderCount = useSelector((state)=> state.loaderOrError.loaderCount)
    const isError =  useSelector((state)=> state.loaderOrError.isError)
    const errorConfig =  useSelector((state)=> state.loaderOrError.errorConfig)

    const dispatch =  useDispatch()
    function startLoader() {
      dispatch(actions.startLoader())
    }

    function stopLoader() {
      dispatch(actions.stopLoader())
    }

    function showError(errorConfigs) {
      dispatch(actions.showErrors(errorConfigs))
    }

    function hideError() {
      dispatch(actions.hideErrors())
    }
    return (
      <OverlayLoaderOrErrorContextProvider value={{ stopLoader, startLoader, showError }}>
        <WrappedComponent {...props} />
        {loaderCount >= 1 && <OverlayLoader isLoading={loaderCount >= 1} />}
        {isError && (
          <OverlayError
            isError={isError}
            errorConfig={errorConfig}
            hideError={hideError}
          />
        )}
      </OverlayLoaderOrErrorContextProvider>
    );
  };
}

export default withOverlayLoaderOrError;
