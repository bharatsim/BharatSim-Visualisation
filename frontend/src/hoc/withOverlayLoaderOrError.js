import React, { useState } from 'react';
import { OverlayLoaderContextProvider } from '../contexts/overlayLoaderContext';
import OverlayLoader from '../uiComponent/OverlayLoader';
import OverlayError from '../uiComponent/OverlayError';
import { errorTypes } from '../constants/loaderAndErrorMessages';

function withOverlayLoaderOrError(WrappedComponent) {
  return (props) => {
    const [loader, setLoader] = useState({ loadersCount: 0 });
    const [error, setError] = useState({
      isError: false,
      errorCode: errorTypes.DEFAULT,
      data: undefined,
      query: undefined,
    });

    function startLoader() {
      setLoader((prevState) => ({
        loadersCount: prevState.loadersCount + 1,
      }));
    }

    function stopLoader() {
      setLoader((prevState) => {
        if (prevState.loadersCount === 0) {
          return { loadersCount: 0 };
        }
        return { loadersCount: prevState.loadersCount - 1 };
      });
    }

    function stopAllLoaders() {
      setLoader({ loadersCount: 0, message: '' });
    }

    function showError(errorCode) {
      stopAllLoaders();
      setError({ isError: true, errorCode });
    }

    function hideError() {
      setError({ isError: false, errorCode: errorTypes.DEFAULT });
    }

    return (
      <OverlayLoaderContextProvider value={{ stopLoader, startLoader, showError }}>
        <WrappedComponent {...props} />
        {loader.loadersCount >= 1 && <OverlayLoader isLoading={loader.loadersCount >= 1} />}
        {error.isError && (
          <OverlayError
            isError={error.isError}
            errorCode={error.errorCode}
            onClose={hideError}
            data={error.data}
          />
        )}
      </OverlayLoaderContextProvider>
    );
  };
}

export default withOverlayLoaderOrError;
