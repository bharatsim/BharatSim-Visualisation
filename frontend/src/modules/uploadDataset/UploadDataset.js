import React, { useContext, useState } from 'react';
import { useHistory } from 'react-router-dom';
import { useSnackbar } from 'notistack';

import { Box, Button, makeStyles, Step, StepLabel, Stepper, Typography } from '@material-ui/core';

import ProjectHeader from '../../uiComponent/ProjectHeader';
import ButtonGroup from '../../uiComponent/ButtonGroup';

import ConfigureDatatype from './ConfigureDatatype';
import ImportDataset from './ImportDataset';
import { projectLayoutContext } from '../../contexts/projectLayoutContext';
import { api } from '../../utils/api';
import { errors } from '../../constants/loaderAndErrorMessages';
import { overlayLoaderOrErrorContext } from '../../contexts/overlayLoaderOrErrorContext';
import DashboardHeaderBar from '../../uiComponent/DashboardHeaderBar';

const useStyles = makeStyles((theme) => ({
  uploadHeader: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: theme.spacing(3, 8),
    width: '100%',
  },
  contentWrapper: {
    backgroundColor: '#FFFFFF',
    border: '1px solid',
    borderColor: `${theme.colors.primaryColorScale['500']}3d`,
    borderRadius: theme.spacing(1),
    margin: theme.spacing(8),
    padding: theme.spacing(8, 10),
  },
}));

function getStepContent(
  stepIndex,
  setFile,
  setSchema,
  file,
  schema,
  handleNext,
  setPreviewData,
  setErrorStep,
  previewData,
) {
  if (stepIndex === 0) {
    return (
      <ImportDataset
        setFile={setFile}
        handleNext={handleNext}
        setPreviewData={setPreviewData}
        setErrorStep={setErrorStep}
        setSchema={setSchema}
      />
    );
  }

  if (stepIndex === 1) {
    return (
      <ConfigureDatatype
        schema={schema}
        selectedFile={file}
        handleNext={handleNext}
        previewData={previewData}
      />
    );
  }

  return (
    <ConfigureDatatype
      schema={schema}
      selectedFile={file}
      handleNext={handleNext}
      previewData={previewData}
    />
  );
}

function UploadDataset() {
  const classes = useStyles();
  const history = useHistory();
  const steps = ['Import Data', 'Preview Data', 'Upload to Dashboard'];

  const {
    selectedDashboardMetadata: { _id: selectedDashboardId, name: selectedDashboardName },
  } = useContext(projectLayoutContext);

  const { showError } = useContext(overlayLoaderOrErrorContext);
  const [previewData, setPreviewData] = useState();
  const [activeStep, setActiveStep] = useState(0);
  const [errorStep, setErrorStep] = useState(undefined);
  const [file, setFile] = useState();
  const [schema, setSchema] = useState();

  const { enqueueSnackbar } = useSnackbar();

  function handleNext() {
    setActiveStep((prevActiveStep) => prevActiveStep + 1);
  }

  function resetAllState() {
    setPreviewData();
    setActiveStep(0);
    setErrorStep(undefined);
    setFile();
    setSchema();
  }

  async function onClickUploadAndSave() {
    handleNext();
    await api
      .uploadFileAndSchema({ file, schema, dashboardId: selectedDashboardId })
      .then(() => {
        history.push(`configure-dataset`);
        enqueueSnackbar(`uploaded ${file.name} to dashboard ${selectedDashboardName}`, {
          variant: 'success',
        });
      })
      .catch((error) => {
        const errorConfigs = errors[error.errorCode](file.name, resetAllState);
        showError(errorConfigs);
      });
  }

  function onCancel() {
    history.push(`configure-dataset`);
  }

  return (
    <Box>
      <ProjectHeader />
      <DashboardHeaderBar>
        <Box className={classes.uploadHeader}>
          <Typography variant="h6"> Configure Dashboard Data :: Upload Dataset</Typography>
          <ButtonGroup>
            <Button variant="text" onClick={onCancel} size="small">
              Cancel
            </Button>
            <Button
              variant="contained"
              color="primary"
              size="small"
              disabled={activeStep !== 1}
              onClick={onClickUploadAndSave}
            >
              Upload
            </Button>
          </ButtonGroup>
        </Box>
      </DashboardHeaderBar>
      <Box className={classes.contentWrapper}>
        <Box px={8} pb={4}>
          <Stepper activeStep={activeStep}>
            {steps.map((label, index) => (
              <Step key={label}>
                <StepLabel error={errorStep === index}>{label}</StepLabel>
              </Step>
            ))}
          </Stepper>
        </Box>
        <Box>
          {getStepContent(
            activeStep,
            setFile,
            setSchema,
            file,
            schema,
            handleNext,
            setPreviewData,
            setErrorStep,
            previewData,
          )}
        </Box>
      </Box>
    </Box>
  );
}

export default UploadDataset;
