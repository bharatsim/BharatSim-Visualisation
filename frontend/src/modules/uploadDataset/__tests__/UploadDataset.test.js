import React from 'react';
import { render } from '@testing-library/react';
import { fireEvent } from '@testing-library/dom';

import * as fileUtils from '../../../utils/fileUploadUtils';
import withThemeProvider from '../../../theme/withThemeProvider';
import { ProjectLayoutProvider } from '../../../contexts/projectLayoutContext';
import UploadDataset from '../UploadDataset';
import { api } from '../../../utils/api';
import withSnackBar from '../../../hoc/withSnackBar';
import withOverlayLoaderOrError from '../../../hoc/withOverlayLoaderOrError';
import { withRouter } from '../../../testUtil';

jest.spyOn(fileUtils, 'parseCsv').mockImplementation((csvFile, previewLimit, onComplete) => {
  const data = { data: [{ col1: 'row1', col2: 1 }], errors: [] };
  onComplete(data);
});

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

jest.mock('../../../utils/api', () => ({
  api: {
    uploadFileAndSchema: jest.fn().mockResolvedValue('success'),
  },
}));

describe('Upload Dataset', () => {
  let ComponentWithProvider;
  beforeEach(() => {
    ComponentWithProvider = withThemeProvider(
      withOverlayLoaderOrError(
        withRouter(
          withSnackBar(() => (
            <>
              <ProjectLayoutProvider
                value={{
                  projectMetadata: { name: 'project1', id: '123' },
                  selectedDashboardMetadata: { name: 'dashboard1', _id: 'dashboardId' },
                }}
              >
                <UploadDataset />
              </ProjectLayoutProvider>
            </>
          )),
        ),
      ),
    );
  });

  it('should match snapshot for upload dataset component', () => {
    const { container } = render(<ComponentWithProvider />);

    expect(container).toMatchSnapshot();
  });

  it('should display import file screen for step 1', () => {
    const { getByText } = render(<ComponentWithProvider />);

    expect(getByText('Drag your file here or')).toBeInTheDocument();
  });

  it('should import file and open configure dataset step ', async () => {
    const { getByText, getByTestId, findByText } = render(<ComponentWithProvider />);
    const inputComponent = getByTestId('file-input');

    fireEvent.change(inputComponent, {
      target: { files: [new File([''], 'testFile.csv', { type: 'text/csv' })] },
    });

    await findByText('DataFile:');

    expect(getByText('testFile.csv')).toBeInTheDocument();
  });

  it('should import file and open configure dataset step and upload data', async () => {
    const { getByText, getByTestId, findByText } = render(<ComponentWithProvider />);
    const inputComponent = getByTestId('file-input');

    fireEvent.change(inputComponent, {
      target: { files: [new File([''], 'testFile.csv', { type: 'text/csv' })] },
    });

    await findByText('DataFile:');

    fireEvent.click(getByText('Upload'));

    await findByText('DataFile:');

    expect(api.uploadFileAndSchema).toHaveBeenCalledWith({
      dashboardId: 'dashboardId',
      file: expect.any(File),
      schema: { col1: 'String', col2: 'Number' },
    });
  });

  it('should navigate to configure-dashboard on click of cancel', async () => {
    const { getByText } = render(<ComponentWithProvider />);

    fireEvent.click(getByText('Cancel'));

    expect(mockHistoryPush).toHaveBeenCalledWith('configure-dataset');
  });

  it('should navigate to configure-dashboard on upload successful', async () => {
    const { getByText, getByTestId, findByText } = render(<ComponentWithProvider />);
    const inputComponent = getByTestId('file-input');

    fireEvent.change(inputComponent, {
      target: { files: [new File([''], 'testFile.csv', { type: 'text/csv' })] },
    });

    await findByText('DataFile:');

    fireEvent.click(getByText('Upload'));

    await findByText('DataFile:');

    expect(mockHistoryPush).toHaveBeenCalledWith('configure-dataset');
  });

  it('should show snackbar for successfully upload data file ', async () => {
    const { getByText, getByTestId, findByText } = render(<ComponentWithProvider />);
    const inputComponent = getByTestId('file-input');

    fireEvent.change(inputComponent, {
      target: { files: [new File([''], 'testFile.csv', { type: 'text/csv' })] },
    });

    await findByText('DataFile:');

    fireEvent.click(getByText('Upload'));

    await findByText('uploaded testFile.csv to dashboard dashboard1');

    expect(getByText('uploaded testFile.csv to dashboard dashboard1')).toBeInTheDocument();
  });

  it('should call show error to show overlay error with error message', async () => {
    api.uploadFileAndSchema.mockRejectedValueOnce({ errorCode: 1008 });
    const { getByText, getByTestId, findByText } = render(<ComponentWithProvider />);
    const inputComponent = getByTestId('file-input');
    const errorMessage = 'File might contain invalid data';

    fireEvent.change(inputComponent, {
      target: { files: [new File([''], 'testFile.csv', { type: 'text/csv' })] },
    });

    await findByText('testFile.csv');

    fireEvent.click(getByText('Upload'));

    await findByText(errorMessage);

    expect(getByText(errorMessage)).toBeInTheDocument();
  });

  it('should reset step to first step of upload file on click of "okay" on error modal', async () => {
    api.uploadFileAndSchema.mockRejectedValueOnce({ errorCode: 1008 });
    const { getByText, getByTestId, findByText } = render(<ComponentWithProvider />);
    const inputComponent = getByTestId('file-input');
    const errorMessage = 'File might contain invalid data';
    const firstStepMessage = 'Drag your file here or';

    fireEvent.change(inputComponent, {
      target: { files: [new File([''], 'testFile.csv', { type: 'text/csv' })] },
    });

    await findByText('testFile.csv');

    fireEvent.click(getByText('Upload'));

    await findByText(errorMessage);

    const errorModalOkayButton = getByText('Okay');

    fireEvent.click(errorModalOkayButton);

    expect(getByText(firstStepMessage)).toBeInTheDocument();
  });
});
