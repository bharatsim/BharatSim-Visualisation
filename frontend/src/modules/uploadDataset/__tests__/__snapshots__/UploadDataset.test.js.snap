// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload Dataset should match snapshot for upload dataset component 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-9"
  >
    <div
      class="MuiBox-root MuiBox-root-12 makeStyles-projectNameBar-10"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5"
      >
        project1
      </h5>
      <button
        class="MuiButtonBase-root MuiButton-root makeStyles-buttonRoot-11 MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Back to recent projects
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiBox-root MuiBox-root-14 makeStyles-dashboardHeaderBar-13"
    >
      <div
        class="MuiBox-root MuiBox-root-15 makeStyles-uploadHeader-7"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
           Configure Dashboard Data :: Upload Dataset
        </h6>
        <div
          class="MuiBox-root MuiBox-root-17 makeStyles-buttonGroup-16"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Cancel
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Upload
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-18 makeStyles-contentWrapper-8"
    >
      <div
        class="MuiBox-root MuiBox-root-19"
      >
        <div
          class="MuiPaper-root MuiStepper-root MuiStepper-horizontal MuiPaper-elevation0"
        >
          <div
            class="MuiStep-root MuiStep-horizontal"
          >
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal"
            >
              <span
                class="MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text"
                    text-anchor="middle"
                    x="12"
                    y="16"
                  >
                    1
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiTypography-root MuiStepLabel-label MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
                >
                  Import Data
                </span>
              </span>
            </span>
          </div>
          <div
            class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
          >
            <span
              class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
            />
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal"
          >
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
            >
              <span
                class="MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text"
                    text-anchor="middle"
                    x="12"
                    y="16"
                  >
                    2
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
                >
                  Preview Data
                </span>
              </span>
            </span>
          </div>
          <div
            class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
          >
            <span
              class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
            />
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal"
          >
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
            >
              <span
                class="MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text"
                    text-anchor="middle"
                    x="12"
                    y="16"
                  >
                    3
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
                >
                  Upload to Dashboard
                </span>
              </span>
            </span>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-20"
      >
        <div
          class="MuiBox-root MuiBox-root-23"
        >
          <div
            class="MuiBox-root MuiBox-root-29 makeStyles-hidden-26"
          >
            <div
              class="MuiBox-root MuiBox-root-30"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 makeStyles-errorTitle-24 MuiTypography-colorError"
              >
                Error:Â 
              </p>
              <p
                class="MuiTypography-root MuiTypography-body2 makeStyles-errorMessage-25 MuiTypography-colorError"
              />
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-31 makeStyles-importDataContainer-21"
          >
            <img
              alt="Import File"
              src="importFileIcon.svg"
            />
            <div
              class="MuiBox-root MuiBox-root-32 makeStyles-textContainer-22"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Drag your file here or 
              </p>
              <div
                class="MuiBox-root MuiBox-root-33"
              >
                <label
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Browse
                    <input
                      accept=".geojson,.csv"
                      data-testid="file-input"
                      style="display: none;"
                      type="file"
                    />
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
